### PR TITLE
feat: add support for linux-musl-arm64

### LINKED ISSUE #8478: Precompiled engine files for `aarch64-unknown-linux-musl` target (Alpine Linux on ARM, e.g. M1)
## Problem

Right now you can't use Prisma on Docker alpine on M1 mac as it requires `aarch64-unknown-linux-musl` binaries. As M1 will see more adoption, people will run into this frequently. 

## Suggested solution
Provide precompiled binaries for `aarch64-unknown-linux-musl` target like we have precompiled binaries for `x86-linux-musl`

## Alternatives
1. Ask user to compile the binary themselves
2. Document use `node:lts` instead of `node:lts-alpine` which is not ideal.

## Additional context
https://github.com/prisma/prisma/issues/7755

Naming as `unknown` is intended like that, see more information internally: https://prisma-company.slack.com/archives/C016KUHB1R6/p1627469574008300?thread_ts=1626873669.001400&cid=C016KUHB1R6

### PR DESCRIPTION
**Pull request by aqrln**
Add support for two new binary targets:

- `linux-musl-arm64-openssl-1.1.x`
- `linux-musl-arm64-openssl-3.0.x`

Checklist:

- [x] Engines build images (https://github.com/prisma/engine-images/pull/66)
   - [x] ready
   - [x] merged
- [x] Add the new target in engineer (https://github.com/prisma/engineer/pull/87)
   - [x] ready
   - [x] merged
- [x] Update engineer in prisma-engines to a tagged version once the engineer PR lands, use the branch for testing meanwhile (https://github.com/prisma/prisma-engines/pull/3656)
   - [x] tested with integration branch
   - [x] release engineer 1.47
   - [x] update prisma-engines to the new engineer
- [x] Add support for the new binary targets on the client (this PR)
- [x] Ecosystem tests (https://github.com/prisma/ecosystem-tests/pull/3418)

Closes https://github.com/prisma/prisma/issues/8478

### CODE REVIEW

File: packages/get-platform/src/getPlatform.ts
Comment (aqrln) [inline packages/get-platform/src/getPlatform.ts:77]: I removed this altogether instead of doing what TODO says because we don't have this check for other platforms, and I don't think it's generally useful to hard error on unsupported CPU architecture because users can totally build from source for 32-bit ARM or RISC-V or whatever they need.

Comment (jkomyno) [reply packages/get-platform/src/getPlatform.ts:77]: Note: we no longer "hard error", we warn now

Comment (aqrln) [reply packages/get-platform/src/getPlatform.ts:77]: Nice, I missed that, thanks for pointing this out!

Comment (aqrln) [reply packages/get-platform/src/getPlatform.ts:77]: Should we still have the check for the CPU architecture and just remove the distro from it then?

**Comment by jkomyno** [comment]
Added ecosystem-tests here: https://github.com/prisma/ecosystem-tests/pull/3418

File: packages/get-platform/src/getPlatform.ts
Comment (jkomyno) [reply packages/get-platform/src/getPlatform.ts:77]: Let's go with `if (platform === 'linux' && !['x64', 'arm64'].includes(arch)) {`

Drop the `alpine:` prefix from the `alpine:unsupported-arch` warning and change the first sentence to `Prisma only officially supports Linux on `amd64` (x86_64) and `arm64` (aarch64) system architectures`, please.

Comment (aqrln) [reply packages/get-platform/src/getPlatform.ts:77]: `platform === 'linux'` check was not necessary here since the whole code path here is only relevant for Linux, other than that, done

**Comment by jkomyno** [comment]
Note: I will approve this PR once https://github.com/prisma/engine-images/pull/66 and https://github.com/prisma/engineer/pull/87 are merged

**Comment by aqrln** [comment]
Tests fail with last integration version (panic in WASM `get_dmmf` in `generator.test.ts`), presumably because the client side now already includes https://github.com/prisma/prisma/commit/8ad76c3e04bd3d037e2258b6219d062d863ee00c after the rebase, however the engine branch didn't include https://github.com/prisma/prisma-engines/commit/323c8a1169351067817dd9f83076b2f8afad856c yet. Since the PR is already merged and the binaries from `main` should be released soon-ish, those should hopefully work.

**Comment by aqrln** [comment]
> those should hopefully work

or rather those should only fail other expected tests since the engine updates are currently blocked on https://github.com/prisma/prisma/pull/17705 / https://github.com/prisma/prisma-engines/pull/3666

**Comment by jkomyno** [comment]
I see failures in `views` functional tests, perhaps that's due to the engines being slightly misaligned with the `prisma-engines`' `main`. After all, views are a preview feature and subject to frequent changes these days.
However, we should investigate and confirm/debunk this.

**Comment by aqrln** [comment]
@jkomyno it’s expected, see the comment above and https://github.com/prisma/prisma/pull/17705

**Comment by jkomyno** [comment]
https://github.com/prisma/prisma/pull/17728 is merged now

**Review by jkomyno — APPROVED** [review]

**Pull request merged**
